client/components/jetpack/card/jetpack-product-card/upgrade-nudge.tsx

```diff
@@ -0,0 +1,144 @@
+/**
+ * External dependencies
+ */
+import classNames from 'classnames';
+import { translate as globalTranslate, useTranslate, TranslateResult } from 'i18n-calypso';
+import { isFinite } from 'lodash';
+import React, { useCallback } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { Button } from '@automattic/components';
+import Gridicon from 'components/gridicon';
+import { preventWidows } from 'lib/formatting';
+import PlanPrice from 'my-sites/plan-price';
+import { JETPACK_OFFER_RESET_UPGRADE_NUDGE_DISMISS } from 'my-sites/plans-v2/constants';
+import { savePreference } from 'state/preferences/actions';
+import { getPreference } from 'state/preferences/selectors';
+import { getPlan } from 'lib/plans';
+
+/**
+ * Type dependencies
+ */
+import { JetpackRealtimePlan } from 'lib/plans/types';
+
+// TODO: replace this hardcoded data with real data when we have an endpoint for it or
+// a more definitive place in our own codebase.
+const DEFAULT_FEATURES = [
+	{
+		title: globalTranslate( 'Real-Time Backup' ),
+		subtitle: globalTranslate( 'Every edit gets saved immediatelly' ),
+	},
+	{
+		title: globalTranslate( 'Real-Time Scan' ),
+		subtitle: globalTranslate( 'On-demand scanning' ),
+	},
+];
+
+type OwnProps = {
+	billingTimeFrame: TranslateResult;
+	className?: string;
+	currencyCode: string;
+	discountedPrice?: number;
+	features?: typeof DEFAULT_FEATURES;
+	onUpgradeClick: () => void;
+	originalPrice: number;
+	productSlug: JetpackRealtimePlan;
+};
+
+const UpgradeNudge = ( {
+	billingTimeFrame,
+	className,
+	currencyCode,
+	discountedPrice,
+	features = DEFAULT_FEATURES,
+	onUpgradeClick,
+	originalPrice,
+	productSlug,
+}: OwnProps ) => {
+	const translate = useTranslate();
+	const isDiscounted = isFinite( discountedPrice );
+
+	const plan = getPlan( productSlug );
```

Comment:
This would fail if the code provides a product, which would happen if we want to nudge for Backup Real-time. This code should be directly passed `ProductSelector`.

Reply:
Good catch. Thanks!